internal/ingress/types.go

```diff
@@ -333,6 +327,12 @@ type Location struct {
 	// Mirror allows you to mirror traffic to a "test" backend
 	// +optional
 	Mirror mirror.Config `json:"mirror,omitempty"`
+	// Policies to describe the characteristics of an alternative backend.
+	// +optional
+	TrafficShapingPolicy TrafficShapingPolicy `json:"trafficShapingPolicy,omitempty"`
+	// Alternative backend name for this location in "Canary" feature
+	// +optional
+	AlternativeBackendName string `json:"alternative-backend-name"`
```

Comment:
Why are you changing this to single alternative backend? The initial plan with `AlternativeBackends` was to support multiple alternative backends and distribute traffic among them. I think we should keep that.